### ISSUE TITLE
[BUG] Map pins not being shared

### ISSUE THREAD

**Issue by pdickey** [valheimPlus/ValheimPlus#672]
- **OS:** Windows Clients, Linux Server
- **Valheim Game Version:** 0.206.5
- **V+ Mod Version:** 0.9.9.7

**Describe the bug**

Map pins are not being shared between players.

**To Reproduce**

Steps for the devs to reproduce the behavior:
1. Enable the map section in the V+ config
2. Enable shareAllPins in the V+ config
3. Launch Valheim and join the server
4. Create a pin. It does not appear to other players on the server.

**Expected behavior**

When shareAllPins is enabled, I expect all map pins to be shared.

**Troubleshooting**

I have restarted both the server and the clients, but the problem persists.

**Additional context**

No other mods are installed besides BepInEx.

**Comment by schloram** [comment]
Any updates on this? I started to use V+ with my Valheim party yesterday with only the [Map] features enabled. But we realized that the shareAllPins option does not work. I read in the V+ Discord that this feature has been retired since 0.9.9.6 and we should use the Cartography Table instead. But unfortunately this is no option for my  party due to inconsistent behavior of the table.

So I just wanted to know if this feature will be back soon or if it is gone for good.

Thanks

**Comment by pdickey** [comment]
If this feature is retired, why is it still in the configs, and why are there no patch notes about retiring it? Also I agree with @schloram that the Cartography Table is hot garbage for sharing pins with a party. V+ worked great for this in 0.9.9.6, and I hope that the V+ team will continue to support this feature.

**Comment by nxPublic** [comment]
**This feature is not retired.**
I was also unaware that the Cartography Table is unreliable.

The only thing I retired is the additional Pin UI that caused so many incompatibility issues with other mods.

I will fix the shareAllPins functions within the next few days (as soon as I humanly can) when I also update V+ for the beta branch content recently released for Valheim.

I'm sorry this hasn't been quicker, I am trying.

**Comment by schloram** [comment]
@nxPublic No need to apologize. :) Just wanted to know If we could expect this feature to be back in the future or not since this feature is actually the only reason (beside Map progression share) why we are using V+.

Great work with V+, mate!

**Comment by pdickey** [comment]
@nxPublic No worries on the delays and I'm very glad to hear that the feature is not retired. Like the other commenter, map progression and pin sharing is the primary reason that my group is using V+, though we've enabled a few other QOL improvements too. Thank you for all of your work on the mod, and no worries that it's taking a little while to implement a fix for this issue.

**Comment by prototype464** [comment]
As an update, this feature is still not working - Using version 0.9.9.8, shared pins aren't working but map exploration sharing is. Nothing seems to have changed?

The removal of the pin UI was appreciated though, as previously it prevented the usage of the toggles for cartography table and pin view on the map couldn't be toggled with the small box in the bottom right of the fullscreen map.

**Comment by schloram** [comment]
@prototype464 As this is still `in process` and the release notes of 0.9.9.8 does not mention this issue I assume it wasn't planned for this release.

I believe the dev will update this issue as soon as this is fixed. Until then we have to be patient :)

**Comment by LetoDK** [comment]
Just chiming in to say that pin sharing is not working for us either. But everything else seems to be working, including getting map data from parts of the map that were explored while another player was not online.
It's just the pin sharing: we haven't been able to see a single of each others' pins.
0.9.9.8

**Comment by snips86x** [comment]
> Just chiming in to say that pin sharing is not working for us either. But everything else seems to be working, including getting map data from parts of the map that were explored while another player was not online. It's just the pin sharing: we haven't been able to see a single of each others' pins. 0.9.9.8

It's known and the feature has been retired while it's worked on.

**Comment by asschun** [comment]
My pin sharing is not working either as of June 28, 2022

**Comment by momocorpo** [comment]
Confirmed not working as of August 11, 2022, any news on a fix for it?

**Comment by pdickey** [comment]
@nxPublic said they would fix the shareAllPins functions within the next few days, but that was back in March. It's been 6 months. I don't think this is a priority for V+, and I'd recommend looking into different mods that have similar functionality instead of continuing to wait on this fix.

**Comment by SSimp258** [comment]
I'm not sure if this will be fixed because the cartography table does this now.

**Comment by pdickey** [comment]
My groups have had two problems with the cartography table. For one, it's not real-time. So if you're exploring an area together, everyone has to drop pins for everything that you might use as landmarks for communication. Then when you use the cartography table, (if it works) you have a slightly different pin for everyone, which is really difficult to clean up. The way V+ did it was cleaner and solved this issue entirely for my groups.

For two, the cartography table is really inconsistent. It seems to work eventually but sometimes it takes multiple in-game days for some teammates to be able to sync the pins that someone else added and synced to the table, while others are able to get them immediately.

Please stop pointing to the cartography table as a solution for this feature which is _still_ documented as part of V+.

**Comment by SSimp258** [comment]
I guess you don't understand how the cartography table works and the way V+ works.

Normally, when you go explore out in the real world, the map back at port doesn't automatically update as you're out there running around. So, the cart table is the real world example on how it should work. V+ makes it convenient as it updates everyone's map in real time which isn't how exploring really works in the real world.

Now, maybe I need to explain how to use the cart table properly:

1 - Assign one person as the pin person. They're the only person that marks stuff if you're moving as a group.
2 - If you're not moving as a group, you need to assign who marks what in a grid pattern. Think like archeology.
3 - Everyone needs to update the cart table every time they come back to the base.
4 - Everyone needs to pull the updated map every time they leave the base.

Not everyone on your server is updating the cart table and that's why it takes 1-2 days for it to update. You got a friend (or even yourself) that's not updating the table and that's why you're complaining that it isn't feasible...but it just comes down to you don't do the steps above. I've never had an issue while using the cart table as i've mentioned so the "well, we update it when we get back" is just cover for "we don't do this religiously."

Now, the fog of war thing should still go away as you move through the unexplored areas and that should update on other's maps instantly if you have it setup that way. It's just the pins aren't sharing and that's really not that big of a deal and why the cart table is the fix. If i had to guess, the map pin sharing part of V+ will go away and will be removed in a major update...but who knows, maybe it'll be fixed in a major update. It's hard saying which way it'll actually go but I will tell you and your players on your server to learn how to use the cart table properly in the meantime just in case it is removed permanently.

**Comment by LetoDK** [comment]
@SSimp258 

It appears you are a bit confused, so let me clear things up. Everyone here understands how the table works (or at least when it actually works). However, Valheim+ provides a map sync feature **that people actually want**, but which sadly is broken at the moment.
You obviously already know this. But just because _you_ don't feel this feature provided additional value compared to the table, it doesn't mean others don't.
Now, I'm sorry but bringing up "_how exploring really works in the real world_" is beyond ridiculous — where're talking about a game where you're a magic viking, who can instantly respawn upon death, who is fighting magical monsters, and xp, and omnipresent world chat, and so on, and so on. We are well beyond "realism". This is not the point.
In fact, we're here because of a mod which fundamentally changes how the game works in many ways, including this valuable sync feature. 
You're arguing that based on the fact that this feature is currently broken, it's actually better without because _you_ like to play without it.
You come here telling others how they should play their game — this in direct contradiction to the whole point of Valheim+, and indeed Valheim itself. This is a game where you can play it in whatever style you prefer.

**Comment by SSimp258** [comment]
The only section that doesn't work is the map pins. The FoW should be working since the last time i played on a server with others. When I was playing the FoW was the issue. No one cared about the map pins and the cartography table took care of that. The workaround for the map pins was the cart table until the function is either fixed or removed completely. It's not up to me to decide if it gets fixed or not. I would gladly fix it but i'm currently in med school so the only thing i can do is look through code and try to figure out where it's broken and even then, this mod isn't programmed in my computer programming language so i'm not all that helpful and since i'm in med school, i don't have the time to learn it.

Also, getting this fixed will basically mean the cart table is obsolete in many ways but one. The pins and FoW will not be shared if other players are offline so that means you'll still need to use the cart table. So that's an argument to just use the cart table even if this gets fixed because there's a good chance people will be offline when someone decides to explore and mark stuff.

**Comment by pdickey** [comment]
> Now, maybe I need to explain how to use the cart table properly:
> 
> 1 - Assign one person as the pin person. They're the only person that marks stuff if you're moving as a group. 
> 2 - If you're not moving as a group, you need to assign who marks what in a grid pattern. Think like archeology. 
> 3 - Everyone needs to update the cart table every time they come back to the base. 
> 4 - Everyone needs to pull the updated map every time they leave the base.
> 
> Not everyone on your server is updating the cart table and that's why it takes 1-2 days for it to update. You got a friend (or even yourself) that's not updating the table and that's why you're complaining that it isn't feasible...but it just comes down to you don't do the steps above. I've never had an issue while using the cart table as i've mentioned so the "well, we update it when we get back" is just cover for "we don't do this religiously."

I'm glad the cartography table works well for you and your groups, but I don't appreciate the insinuation that me and my friends don't know how to use a 2-button device that basically equates to "save" and "load." I'll explain for you since you seem to think that I need basic instructions. I have done the following testing:

1 - Add a new pin on my map
2 - Record my discoveries to the table
3 - Have a friend try to read discoveries from the table. They get the message "There is nothing new to copy"
4 - Have friend record their discoveries and retry step 3. They get the same message: "There is nothing new to copy"
5 - Eventually, the friend will go to use the table and for no apparent reason it will work and they'll get the pin.

We know how to use the cartography table, but it does not work reliably and the pin sharing is not real-time which creates problems for us during group exploration.

> Also, getting this fixed will basically mean the cart table is obsolete in many ways but one. The pins and FoW will not be shared if other players are offline so that means you'll still need to use the cart table. So that's an argument to just use the cart table even if this gets fixed because there's a good chance people will be offline when someone decides to explore and mark stuff.

Why is it bad if V+ obsoletes the cartography table? You can use V+ to modify many other core elements of the game. The point of V+ is to let you play the game the way you and your group want to play. My group explores together because that's how we want to play. People might jump on to do some solo work on one of our bases, or tend crops, or organize storage, but we almost exclusively go exploring out into the world as a group. We would not need a cartography table at all if V+ worked the way it says it should.

**Comment by SSimp258** [comment]
This is why V+ is configurable to how you want it set up. I'm for it being fixed even if i don't use it. I'm just sayin the cart table is the fix for now whether it gets fixed or not. Me and my friends never had an issue with the cart table even as you described which leads me to believe you guys don't interact with it all the time so the "save" and "load" like you say isn't happening correctly. Me and my friends interact with it all the time before and after we leave our base. I just don't like the way the table saves pins if more than 1 person marked something. It puts X amount of pins depending on how many people marked it.

I understand your frustration but people have RL stuff to do so working on a mod that's free can take the back burner a lot of the time. I'm in med school in the middle of finals so i stick to mostly discord troubleshooting so people can at least play with V+. I have a BS degree in Software Development in Java so i'm not much help with coding and it'd take me a couple of weeks to learn the code to a decent level to help with coding fixes or add features that people would like. I wouldn't mind expanding on the chat to make it more like an IRC chat so you don't have to face the person when they shout or what not but i just don't have the time right now to try to do stuff like that. I know NX had some RL stuff they've been dealing with the last couple of months and i don't really interact with the other devs helping NX that much.

I try to tell people if you wanna help out, fork the Git and program stuff. If you can't code, learn to troubleshoot the issues that people have reported and then come in here and post what you've done to try and narrow down what's wrong. You have no idea how much help that is when trying to look for errors in the code. I had asked someone in discord to try a test and we found out that moving a chest with AEM breaks craft from chests but if you place the chest with ABM, it's fine because the code checks how the chest was placed..by the player or not. Because it does a check in the code and it would appear there's a flag for either the player or the system and when using AEM, the chest is flagged by the system and not the player so the check for building from chests is ignored because the player didn't place it, the system did.

**Comment by LetoDK** [comment]
@SSimp258 just to be clear, you're in med school, right? I just wasn't sure. 

Anyways, I would like to point out that nobody is complaining. We're just reporting bugs. You're the one who came into a thread about a specific bug and went full "it's not a bug; it's a feature" on behalf of devs that didn't ask you to do so.

**Comment by SSimp258** [comment]
I had a whole thing typed but i deleted it. I'll just say it'll be handled whenever NX or the devs get to it. They know it's an issue. And yes, if you were wondering I'm a chat moderator in the discord and we've talked about this stuff in the past and whether or not these types of things get fixed or not. Right now, the fix or "temp fix," if you will, is to use the cart table. We have no control over the cart table or if it works 110% of the time for everyone. Waiting 3 months and coming in here and posting on a bug to make it relevant again actually doesn't help. If you want to help, post some troubleshooting steps you took to narrow down the issue because that helps the devs out a ton to identify the issue in the code. The other thing you can do, if you can code, is to fork the Github and fix it yourself. I have a final on Monday that i'm studying for. After my test, i have a couple weeks break so i can look at it then but i can't promise it'll be fixed because I don't know the code.

**Comment by pdickey** [comment]
> post some troubleshooting steps you took to narrow down the issue

You mean, like I did in the original post? I'm not sure what other troubleshooting you'd want anyone to do for this issue. There are no other mods, the config is enabled, and it doesn't work. It's not intermittent at all. This broke when NX took out the map pin UI, so it's likely related to that, but that's also obvious from the original context. If you have suggestions other than "read the code and fix it yourself" I'm all ears.

PS. The repository also has some pretty old pending pull requests. Not sure what any of those are doing or trying to fix, but it doesn't give me confidence that if I did spend the time to fix this myself that my patch would be applied in a timely fashion.

**Comment by SSimp258** [comment]
If you think it's from the UI change thing that gives me a place to start looking. Maybe I can just redo the UI and make it work which would be the easiest thing to do especially if it was working before the UI change.

**Comment by LetoDK** [comment]
> Waiting 3 months and coming in here and posting on a bug to make it relevant again actually doesn't help.

Actually, that's not true. Many devs on side-projects have a lot of things they could potentially work on, but have limited time to do so. Therefore, getting reports is a way for us to see what features are most wanted, and what problems have what kinds of persistence. This is all data that can help with prioritizing efforts and directions. 

Also, I don't know what a given dev is working on at given moment, and I don't have time to check up on the whole history of a given work (I'm busy most of the year, including the weeks where some are doing their finals).

> post some troubleshooting steps

Ah, I now see you haven't read this thread (except maybe your own comments). There are no steps; the pins are not syncing. That's it. Step 0, done.

> After my test, i have a couple weeks break so i can look at it then but i can't promise it'll be fixed because I don't know the code.

You can do that if you want to, but if I were you, I would enjoy my break instead of trying to wrap my head around some code that I have no experience with, in another language than I was once used to. The devs will fix it if they get the time, energy, and impulse. And if that happens, then those of us who reported the bug will be grateful, and if not, then too bad, such is life. But sometimes a bug can never get fixed if people don't report it.

Good luck with your finals.

**Comment by MartenBE** [comment]
After quite a while, this issue doesn't seem solved. What do you guys do to play without this feature?
I would like to sink my teeth into this, but I only have a Java / C++ background and know nothing from C#, Unity, HarmonyX etc. If a dev whishes to bring me up to speed, I'd like to take a shot at it ...

**Comment by SerpentDrago** [comment]
Reporting its still broken . I would love to see this feature . I don't' know of any currently updated mod that supports syncing of map pins across a server .

**Comment by SSimp258** [comment]
So this feature has been retired. Iron Gate has added a cartography table that does the same thing. I believe you need to kill the 1st boss to have access to a mining pickaxe and i think it takes either copper or bronze to make the table.

**Comment by LetoDK** [comment]
@SSimp258
> So this feature has been retired.


You have been falsely claiming this for months on the basis of your lack of understanding regarding the substantial and meaningful differences between this specific (currently bugged) feature and the cartography table. We have tried to explain to you how these are significant differences, but to no avail. 

So let's once again go back to _the words of nxPublic_:

**"This feature is not retired."**

**Comment by snips86x** [comment]
> @SSimp258
> 
> > So this feature has been retired.
> 
> You have been falsely claiming this for months on the basis of your lack of understanding regarding the substantial and meaningful differences between this specific (currently bugged) feature and the cartography table. We have tried to explain to you how these are significant differences, but to no avail.
> 
> So let's once again go back to _the words of nxPublic_:
> 
> **"This feature is not retired."**

Unfortunately you are incorrect. 29/01/2022, he stated on the discord server "the code is not removed, just retired until I can revisit".
So yes, it has been retired.

**Comment by LetoDK** [comment]
> until I can revisit

Understand that the word "retire" has more meanings than "permanent retirement". For example, you can say "I will retire" when going to bed.

The quote you just shared is clearly trying to communicate "I have paused work on this specific feature for now, to work on other stuff" as opposed to "this feature has been removed", (in fact it literally says it _isn't_ removed). These are very different messages.

**Comment by SSimp258** [comment]
Here's my question to you LetoDK:

Why waste time and effort on something when there's already a process for it that's probably better than what you had? Why reinvent the wheel? Why have two processes that does almost THE EXACT SAME THING? (The only difference being how the UI does it)

**Comment by LetoDK** [comment]
I'm not going to explain to you the differences between a screw driver and a hammer yet again. See my previous explanations for some bedtime reading.

It's not really relevant that differences between these two tools weren't obvious is your play style. To a lot of us, these two different tools, with the associated  differences in functioning, served different purposes (as different tools tend to do, regardless of surface level similarities). I can't really do much to put it more simply for you.

**Comment by SSimp258** [comment]
So my take away is this:

You're just too lazy to use the cartography table.

Noted. But just an FYI, NX will probably not come back to it since Iron Gate added the cartography table. Happy Gaming!

**Comment by pdickey** [comment]
Resolved: won't fix

**Comment by pdickey** [comment]
Dev clearly does not plan to fix this. Closing issue.

Subsequent issues should be opened to correct the documentation that claims this is a feature of valheimPlus

**Comment by MartenBE** [comment]
What a toxic shitshow these last posts where.

@SSimp258 isn't even a contributor (no contributions to the code in anyway) so he's just trolling, @nxPublic said he was looking into it although it won't be for quite a while which is ok for me. Last time I checked, the "in progress" label is still applied here.

Yes, there is a cartography table. I just like to make life easier whilst managing pins with 10 people without the table bugging out. I am lazy, sure. That's among other things what this mod is for, to make it all a bit easier so sue me :) . Also, the cartography table sometimes leads to loss of pins: the way it works (only store the last pin layout) makes it vulnerable to race conditions ( https://en.wikipedia.org/wiki/Race_condition#In_software ). In order to get everyone's pins, you'll need one person to update the map, everyone touches the table, someone else updates the map, everyone touches the table, and so on, which is difficult with 5+ players to get this done well and not messing it up. It's very much like using google drive for programming tasks instead of git: perhaps managable with few, cumbersome and dangerous with many.

In short:

- If you would like the feature to be implemented: lets keep this issue open and let us look into it: I'd like to take a stab at it.

- If you don't like the feature to be implemented, just leave this topic alone and do something else with your time. Nothing will change for you even if it would be implemented, just like everything else it would be optional.

So @pdickey, lets reopen this issue. if I create a branch, would you like to help me develop or test? @SSimp258 perhaps you just leave this alone and do something else as it wont influence you in any way? Would a contributor like @healiha be interested in shining their light on this and cleaning this thread a bit so we can start being constructive again?

**Comment by SSimp258** [comment]
There's a small issue with your assumption. I'm a chat mod on the discord and i'm privy to discussions that you are not. I would also contribute but my issue is i'm in medical school and the devs on this project know that. A lot of my time is studying. That means I don't have time to look into a feature that has become obsolete because of the cartography table. So, again, let me say "Happy Gaming" while I leave to go take my Neurology NBME test this morning. kthnxbai!

**Comment by pdickey** [comment]
@MartenBE this issue was open for almost a year with no apparent progress. Its clear that this is not a priority for the developers. From my view, they're burned out on this project. That's certainly understandable, but it's time to give up on this feature.

@SSimp258 I'm glad to hear you're still in medical school. You must read your textbooks more attentively than you read GitHub issues. You are toxic AF. I hope your schooling teaches you to listen to people before we end up reading a bunch of 1-star reviews on healthgrades because you decided their problem wasn't really a problem.

**Comment by LetoDK** [comment]
@pdickey I would recommend blocking the "Did I mention I'm in med school?"-troll before he comes back here to troll some more (apparently he's not too busy with mEd sChOoL to comment extensively on GitHub issues he doesn't fully comprehend — luckily the "I have to be right even when I'm wrong"-attitude is a great character trait for an upcoming medical doctor😒).

**Regarding the issue**, I understand why you might have lost faith in the devs finding the time or will to  fix this bug, but it's still an active and relevant bug regardless, so we might as well keep it open, IMHO).

**Comment by SerpentDrago** [comment]
Yeh you don't close a issue just cause the dev's haven't had the time to fix it . Issues are NOT just for the current devs but also for Reference for any future devs AND forks that may come about .  if you feel you need to clean up the issues you use tags and such

**Comment by SSimp258** [comment]
So let me get this straight. I’m the toxic person? I’m the person getting insulted left and right. I’m the one getting verbally abused by a bunch of Karen’s and Kevin’s DEMANDING stuff get fixed. How are you guys contributing exactly? I understand why 1 or 2 devs left the project now. People like you guys demanding stuff and then playing the victim when someone tells you the feature is retired. 1 of the other chat mods came in here with a previous message from NX that was in discord and told you guys but nah, that wasn’t enough. Let’s attack this person on Git…maybe that’ll spur the devs to fix this. I spent 12 years in the military and dealt with way worse toxic people than you guys. Try harder.

If you guys wanna try and fix it, I’m NX would appreciate it. What I was told when I looked at it when on break was the pin UI needed a complete rewrite and redesign because NX didn’t like it. Hopefully that’ll give you guys a starting point.

Good Luck,

Steph

PS - don’t copy anyone’s code or NX won’t approve the merge.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by MartenBE** [comment]
Can we just stop this please and focus on the issue and how we could help to solve it?
We'll create a fork, try stuff out, and if it works we'll create a pull request back and everyone's happy. No one can argue with that, I think.

**Comment by pdickey** [comment]
For everyone suggesting I reopen this issue: The only update from anyone with a role in maintaining the code was 297 days ago, where they stated they would try to fix this in the next few days. Additionally, discord mods continually derail the thread anytime a new person comments to say they are having the same issue, and instead tell them to use the cartography table. I've closed the issue as not planned as I believe that to be an accurate status. Maintainers on the repository should have sufficient permissions to override my decision if they disagree with me here.

@SSimp258 You are in here directly contradicting [the words of a maintainer](https://github.com/valheimPlus/ValheimPlus/issues/672#issuecomment-1046189120), disregarding the many [places](https://github.com/valheimPlus/ValheimPlus/issues/672#issuecomment-1046082376) where [people](https://github.com/valheimPlus/ValheimPlus/issues/672#issuecomment-1213379225) have [explained](https://github.com/valheimPlus/ValheimPlus/issues/672#issuecomment-1213475293) why they [want this feature](https://github.com/valheimPlus/ValheimPlus/issues/672#issuecomment-1213427459) and why the cartography table doesn't work for them, and you're responding to tell them to use the cartography table. That's toxic behavior.

**Comment by SSimp258** [comment]
> @SSimp258 You are in here directly contradicting the words of a maintainer disregarding the many places where people have explained why they want this feature and why the cartography table doesn't work for them, and you're responding to tell them to use the cartography table. **That's toxic behavior.**

So, my understanding of what toxic behavior is someone coming in here and telling you something is retired because that's what i've been told and have read in a lot of the discord messages.

So, What do you call "name calling, insulting a discord moderator(for no reason) and verbally abusing that discord moderator?"

www.psychcenteral.com says:

What is a toxic personality?
Often the phrase “toxic person” is used to describe someone who is **_subtly or outwardly manipulative_**, **_self-centered_**, **_needy, or controlling._** 

Another blog says:

Toxic behavior definition refers to a person whose behavior and ongoing actions **_cause harm to other people by physical or mental means._** These are the people who spread negative or toxic stress or traumas on others behavior.

So, by these definitions, YOU GUYS ARE THE TOXIC PARTY. You guys are hurling insults, verbal abuse because i came in here to tell you it's been retired. That's all I did and BOOM! You guys went off on me. Wow. You guys must not understand what you're actually doing causing mental harm to me. Please get off Git. Your intentions were to help, at least in the beginning, but you're not helping now, and you guys should just get off Git.

**Comment by MartenBE** [comment]
@pdickey I understand, this issue is going nowhere, so I'm unsubscribing myself from this issue ... I've created a fork and will let you know if something comes out of it. I see where the original code is (it literary says "_Below is the code for the retired Vplus pin share system, i (nx) plan on working with this a little more when i get to it and have a plan on how i would like to change this. This will require some more days/weeks until i get to it, sorry._"), so I don't think @nxPublic will mind if I try something out (if he wants to share me his plan, even better hint hint :) ). Just the debugging will be a pain in the ass I think, no clue on how to do that without multiple players ...

**Comment by LetoDK** [comment]
>I've created a fork and will let you know if something comes out of it.

@MartenBE Great stuff!

> Just the debugging will be a pain in the ass I think, no clue on how to do that without multiple players ...

I would definitely like to help out, if we can manage to find some time where we are both/all available. I'm in central Europe, timezone CET. Otherwise, I regularly play using VH+ with two other friends, and so we can test it out and report back.
(Sorry if this ping brings you back to this thread against your intentions).

@pdickey  Please please consider blocking any accounts that consistently behave in rather unhelpful ways, as that will prevent them from commenting on this issue (if I'm reading the GitHub page correctly)

> After you've blocked a user, they cannot:
>[React to your comments on issues, pull requests, and commits](https://docs.github.com/en/communities/maintaining-your-safety-on-github/blocking-a-user-from-your-personal-account)

**Comment by pdickey** [comment]
@MartenBE Good luck! There is another open issue about this where you may find some people willing to help you test a fix once you are at that stage. As I understand it, NX disabled the map pin UI because there were conflicts between that UI and other mods.

@LetoDK done and thanks for the tip! Now I just wish I had some way to clean up this thread

**Comment by healiha** [comment]
Hello guys,

I've been following what was said here and just to let you know that I'll do my best to work on this by EOY.
I'm currently underwater due to work but this is on my to-do list and I hope to have something to share with you soon.

Please bear with us for the time being, the issue might have been closed but I'll do what I can to make the sharing pins working again when I have time to put on V+.

**Comment by pdickey** [comment]
@healiha Would you like me to reopen the issue then?

**Comment by b-brefeld** [comment]
Sorry for the bump but I installed Valheim Plus today and the things I enabled seem to work fine (except me requiring to outcomment a line of config to get it to work. (Game on steam, Mod through Vortex, let me know if you need more info)).

However, I noticed none of the map sharing features seem to be working.
I'm not playing on a dedicated server but simply starting a server through the game and having my friend join me who is on the PC next to me LAN.

This is the related config snippet that doesn't seem to work:
```
[Map]

; Change false to true to enable this section. https://valheim.plus/documentation/list#Map
enabled = true

; With this enabled you will receive the same exploration progression as other players on the server.
; This will also enable the option for the server to sync everyones exploration progression on connecting to the server.
shareMapProgression = true

; The radius of the map that you explore when moving.
exploreRadius = 100

; Prevents you and other people on the server to turn off their map sharing option.
preventPlayerFromTurningOffPublicPosition = true

; This option automatically shares created pins with everyone playing on the server.
shareAllPins = true

; Display carts and boats on the map
displayCartsAndBoats = true
```

Maybe I'm missing something obvious but I couldn't fine more info on the website documentation or by looking around.
I also made sure to test if it was fixed when we had a "cartography table" but it made no difference.
Is this still still a known bug that isn't fixed or am I doing something wrong.
If it's still an open bug, is there anything I can do to help?
I'm a developer myself even though my experience with C# is limited.
